packages/client/src/runtime/core/model/aggregates/groupBy.ts

```diff
@@ -10,13 +11,15 @@ import { desugarUserArgs as desugarUserArgsAggregate } from './aggregate'
  * @param userArgs to transform
  * @returns
  */
-function desugarUserArgs(userArgs: object) {
+function desugarUserArgs(userArgs: UserArgs) {
   const _userArgs = desugarUserArgsAggregate(userArgs)
 
   // we desugar the array into { [key]: boolean }
-  if (Array.isArray(userArgs?.['by']) === true) {
-    for (const key of userArgs?.['by'] ?? []) {
-      _userArgs['select'][key] = true
+  if (Array.isArray(userArgs['by'])) {
+    for (const key of userArgs['by']) {
+      if (typeof key === 'string') {
+        _userArgs['select'][key] = true
+      }
```

Comment:
Seems that `if (typeof key === 'string') {` is new here, what is its purpose in this context? (add a comment maybe?)

Reply:
It's to keep the checker happy (it was `any` before).